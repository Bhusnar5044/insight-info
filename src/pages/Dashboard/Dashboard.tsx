import DashboardContent from '@components/DashboardContent';
import { Box } from '@components/common';
import { FC, memo } from 'react';

export const Dashboard: FC = memo(() => {
    return (
        <>
            <Box display="flex" flexDirection="column" alignItems="center" gap="2rem">
                {/* <Todo /> */}
                <DashboardContent />
            </Box>
        </>
    );
});

Dashboard.displayName = 'Dashboard';
