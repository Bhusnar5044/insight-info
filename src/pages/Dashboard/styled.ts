import { Box } from '@components/common';
import { styled } from '@utils';

export const StyledContainer = styled(Box)`
    width: 100%;
`;
