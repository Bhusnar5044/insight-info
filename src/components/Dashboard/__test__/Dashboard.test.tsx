import { render, screen } from '@test-utils';
import { Dashboard } from '../Dashboard';

const renderer = () => render(<Dashboard />);
describe('Dashboard', () => {
    it('should render properly', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should render with tabs', () => {
        renderer();
        expect(screen.getAllByText('Applications')).toHaveLength(2);
        expect(screen.getByText('Resources')).toBeInTheDocument();
    });
});
