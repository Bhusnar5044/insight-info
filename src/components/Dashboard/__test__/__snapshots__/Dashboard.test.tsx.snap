// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
}

.c8 {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.c13 {
  display: flex;
  gap: 0.5rem;
}

.c12 {
  -webkit-appearance: none;
  appearance: none;
  width: 100%;
  border: 0;
  font-family: inherit;
  font-size: 16px;
  font-weight: 400;
  background: #ffff;
  box-shadow: inset 0 -1px 0 rgba(#212529, 0.3);
  color: #212529;
  transition: all 0.15s ease;
  outline: none;
}

.c10 {
  position: relative;
  border-radius: 3px;
  padding: 0.5rem 1rem;
  display: flex;
  align-items: center;
  overflow: hidden;
  width: 23rem;
  background-color: #ffff;
}

.c10 .focus-bg {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(#6c757d, 0.05);
  z-index: -1;
  transform: scaleX(0);
  transform-origin: left;
}

.c10 .c11 {
  width: 20rem;
}

.c10 img {
  cursor: pointer;
  width: 1.2rem;
  height: 1.2rem;
}

.c3 {
  display: flex;
  align-items: center;
  list-style: none;
  padding: 10px 0 30px;
}

.c4:not(:last-child) {
  margin-right: 20px;
}

.c5 {
  background: rgba(0, 0, 0, 0);
  padding: 4px;
  color: #ffff;
  box-shadow: none;
  outline: none;
  border: none;
  cursor: pointer;
  font-size: 20px;
  display: block;
  position: relative;
  font-family: inherit;
  font-weight: 400;
}

.c5:after {
  content: '';
  display: block;
  position: absolute;
  left: 0;
  bottom: -2px;
  width: 100%;
  height: 2px;
  background: #ff6e6c;
  opacity: 0;
  visibility: hidden;
  transition: visibility 0s,opacity 0.4s ease-in-out;
}

.c5.active::after {
  opacity: 1;
  visibility: visible;
  transition: visibility 0s,opacity 0.4s ease-in-out;
}

.c6 {
  padding: 2rem;
  color: #c9bae2;
  font-family: inherit;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.4;
  background: #463366;
}

.c2 {
  width: 100%;
}

.c7 {
  margin: 0;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 400;
  letter-spacing: -0.04rem;
  line-height: 3.2rem;
  text-align: initial;
}

.c14 {
  width: 10rem;
  height: auto;
  margin: auto;
  animation: gOaCMp infinite 20s linear;
}

.c1 {
  width: 90%;
  min-height: 90vh;
  margin: 2rem auto;
}

.c9 {
  margin: 2rem 0;
}

<div>
  <div
    class="c0 c1"
    flexdirection="column"
  >
    <div
      class="c0 c2"
      flexdirection="column"
    >
      <ul
        class="c3 tabs__nav"
      >
        <li
          class="c4 tabs__item"
        >
          <button
            class="c5 tabs__TaButton active"
          >
            Applications
          </button>
        </li>
        <li
          class="c4 tabs__item"
        >
          <button
            class="c5 tabs__TaButton "
          >
            Resources
          </button>
        </li>
      </ul>
      <div
        class="c0 c6"
        flexdirection="column"
      >
        <h3
          class="c7"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
        >
          Applications
        </h3>
        <div
          class="c8 c9"
          flexdirection="column"
          gap="1rem"
        >
          <div
            class="c10"
            for="searchInput"
          >
            <input
              class="c11 c12"
              id="searchInput"
              placeholder="Search"
              type="text"
              value=""
            />
            <div
              class="c13"
              gap="0.5rem"
            >
              <img
                alt="search-icon"
              />
            </div>
          </div>
          <img
            class="c14"
            title="loader"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
